### ISSUE TITLE
Consider extracting functions to Kotlin extension

### ISSUE THREAD

**Issue by hoangnguyen92dn** [miles-pudge-halter/nimble-survey-android#5]
With Kotlin programming language, `Extension` is a wonderful choice for us to extend the class with the new functionality rather than create a new class to define more functions. I'd suggest extracting the functions to extensions instead:

- [ ] The extension to validate the `String` is an email or not

https://github.com/miles-pudge-halter/nimble-survey-android/blob/66a7588f2780826d06c4b4b4f315f59e85b495dd/common/src/main/java/me/ayoe/common/EmailValidator.kt#L6-L14

- [ ] The `mapper` classes can be replaced by the extensions to map the `Response` classes to model

https://github.com/miles-pudge-halter/nimble-survey-android/blob/66a7588f2780826d06c4b4b4f315f59e85b495dd/data/src/main/java/me/ayoe/data/mapper/SurveyMapper.kt#L13-L27

P/s: The `let` on the block of code above is useless 🗑️

**Comment by miles-pudge-halter** [comment]
Yes. The email validator should be converted to an extension function like `String.validateEmail()`

The mapper class is an architectural choice to have all the mappings of models under the same title to be under the same class. Projects I worked on before sometimes include similar models with different attributes and some cross model mappings too.

The `let` part of the block should include this. A newbie mistake 🙏

```
questions = response.questions?.let {
                it.map { q ->
                    mapQuestionResponseToDomain(q)
                }
            } ?: listOf()
```

**Comment by hoangnguyen92dn** [comment]
> The mapper class is an architectural choice to have all the mappings of models under the same title to be under the same class. Projects I worked on before sometimes include similar models with different attributes and some cross model mappings too.

Yeah. But my suggestion is something like this: `Respone.toModel() = Model()`, then we could get rid of the Mapper classes.

> The let part of the block should include this. A newbie mistake 🙏

Actually, it could be shorten like this 😄 
```
questions = response.questions?.map(::mapQuestionResponseToDomain).orEmpty()
```
